### PR TITLE
Add ability to set EKS properties on batch

### LINKED ISSUE #29383: [Enhancement]: AWS Batch Job Definition property, eksProperties, missing
### Terraform Core Version

1.3.5

### AWS Provider Version

4.54.0

### Affected Resource(s)

AWS Batch Job Definition functionality of provisioning jobs inside an EKS cluster is missing.
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-batch-jobdefinition.html -> eksProperties should exist but does not within the Terraform AWS Provider

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/batch_job_definition

### Expected Behavior

able to provision resource of AWS Batch Job Definition with eksProperties

### Actual Behavior

eksProperties not found

### Relevant Error/Panic Output Snippet

_No response_

### Terraform Configuration Files

any string value for resource's eksProperties, the TF client will throw an error that property is not found.

### Steps to Reproduce

these steps https://docs.aws.amazon.com/batch/latest/userguide/getting-started-eks.html
- namespace in k8s
- clusterRole
- clusterRoleBinding
- role
- roleBinding
- run eksCtl command mentioned in documentation to authenticate into eks cluster
- create AWS Batch ComputeEnvironment
- create AWS Batch Job Queue
- create AWS Batch Job Definition with eksproperties -> runtime will fail property not found

### Debug Output

_No response_

### Panic Output

_No response_

### Important Factoids

_No response_

### References

_No response_

### Would you like to implement a fix?

None

### PR DESCRIPTION
**Pull request by danquack**
### Description

This PR unlocks a test case for #34663.

### Relations

Closes #29383

### Output from Acceptance Testing

```console
% make testacc TESTS=TestAccBatchJobDefinition_ PKG=batch
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/batch/... -v -count 1 -parallel 20 -run='TestAccBatchJobDefinition_'  -timeout 360m
=== RUN   TestAccBatchJobDefinition_basic
=== PAUSE TestAccBatchJobDefinition_basic
=== RUN   TestAccBatchJobDefinition_disappears
=== PAUSE TestAccBatchJobDefinition_disappears
=== RUN   TestAccBatchJobDefinition_PlatformCapabilities_ec2
=== PAUSE TestAccBatchJobDefinition_PlatformCapabilities_ec2
=== RUN   TestAccBatchJobDefinition_PlatformCapabilitiesFargate_containerPropertiesDefaults
=== PAUSE TestAccBatchJobDefinition_PlatformCapabilitiesFargate_containerPropertiesDefaults
=== RUN   TestAccBatchJobDefinition_PlatformCapabilities_fargate
=== PAUSE TestAccBatchJobDefinition_PlatformCapabilities_fargate
=== RUN   TestAccBatchJobDefinition_ContainerProperties_advanced
=== PAUSE TestAccBatchJobDefinition_ContainerProperties_advanced
=== RUN   TestAccBatchJobDefinition_updateForcesNewResource
=== PAUSE TestAccBatchJobDefinition_updateForcesNewResource
=== RUN   TestAccBatchJobDefinition_tags
=== PAUSE TestAccBatchJobDefinition_tags
=== RUN   TestAccBatchJobDefinition_propagateTags
=== PAUSE TestAccBatchJobDefinition_propagateTags
=== RUN   TestAccBatchJobDefinition_ContainerProperties_EmptyField
=== PAUSE TestAccBatchJobDefinition_ContainerProperties_EmptyField
=== RUN   TestAccBatchJobDefinition_NodeProperties_basic
=== PAUSE TestAccBatchJobDefinition_NodeProperties_basic
=== RUN   TestAccBatchJobDefinition_NodePropertiesupdateForcesNewResource
=== PAUSE TestAccBatchJobDefinition_NodePropertiesupdateForcesNewResource
=== RUN   TestAccBatchJobDefinition_EKSProperties_basic
=== PAUSE TestAccBatchJobDefinition_EKSProperties_basic
=== RUN   TestAccBatchJobDefinition_EKSProperties_update
=== PAUSE TestAccBatchJobDefinition_EKSProperties_update
=== RUN   TestAccBatchJobDefinition_createTypeContainerWithBothProperties
=== PAUSE TestAccBatchJobDefinition_createTypeContainerWithBothProperties
=== RUN   TestAccBatchJobDefinition_createTypeContainerWithNodeProperties
=== PAUSE TestAccBatchJobDefinition_createTypeContainerWithNodeProperties
=== RUN   TestAccBatchJobDefinition_createTypeMultiNodeWithBothProperties
=== PAUSE TestAccBatchJobDefinition_createTypeMultiNodeWithBothProperties
=== RUN   TestAccBatchJobDefinition_createTypeMultiNodeWithContainerProperties
=== PAUSE TestAccBatchJobDefinition_createTypeMultiNodeWithContainerProperties
=== CONT  TestAccBatchJobDefinition_basic
=== CONT  TestAccBatchJobDefinition_ContainerProperties_EmptyField
=== CONT  TestAccBatchJobDefinition_createTypeContainerWithBothProperties
=== CONT  TestAccBatchJobDefinition_updateForcesNewResource
=== CONT  TestAccBatchJobDefinition_PlatformCapabilitiesFargate_containerPropertiesDefaults
=== CONT  TestAccBatchJobDefinition_createTypeContainerWithNodeProperties
=== CONT  TestAccBatchJobDefinition_PlatformCapabilities_fargate
=== CONT  TestAccBatchJobDefinition_PlatformCapabilities_ec2
=== CONT  TestAccBatchJobDefinition_createTypeMultiNodeWithContainerProperties
=== CONT  TestAccBatchJobDefinition_ContainerProperties_advanced
=== CONT  TestAccBatchJobDefinition_propagateTags
=== CONT  TestAccBatchJobDefinition_tags
=== CONT  TestAccBatchJobDefinition_EKSProperties_basic
=== CONT  TestAccBatchJobDefinition_EKSProperties_update
=== CONT  TestAccBatchJobDefinition_NodePropertiesupdateForcesNewResource
=== CONT  TestAccBatchJobDefinition_NodeProperties_basic
=== CONT  TestAccBatchJobDefinition_disappears
=== CONT  TestAccBatchJobDefinition_createTypeMultiNodeWithBothProperties
--- PASS: TestAccBatchJobDefinition_createTypeContainerWithNodeProperties (10.20s)
--- PASS: TestAccBatchJobDefinition_createTypeMultiNodeWithBothProperties (10.20s)
--- PASS: TestAccBatchJobDefinition_createTypeMultiNodeWithContainerProperties (10.37s)
--- PASS: TestAccBatchJobDefinition_createTypeContainerWithBothProperties (10.38s)
--- PASS: TestAccBatchJobDefinition_disappears (24.99s)
--- PASS: TestAccBatchJobDefinition_propagateTags (25.07s)
--- PASS: TestAccBatchJobDefinition_PlatformCapabilities_ec2 (27.47s)
--- PASS: TestAccBatchJobDefinition_ContainerProperties_advanced (28.64s)
--- PASS: TestAccBatchJobDefinition_basic (29.24s)
--- PASS: TestAccBatchJobDefinition_NodeProperties_basic (29.44s)
--- PASS: TestAccBatchJobDefinition_ContainerProperties_EmptyField (29.47s)
--- PASS: TestAccBatchJobDefinition_PlatformCapabilities_fargate (29.60s)
--- PASS: TestAccBatchJobDefinition_PlatformCapabilitiesFargate_containerPropertiesDefaults (29.62s)
--- PASS: TestAccBatchJobDefinition_EKSProperties_basic (29.77s)
--- PASS: TestAccBatchJobDefinition_updateForcesNewResource (36.71s)
--- PASS: TestAccBatchJobDefinition_EKSProperties_update (36.97s)
--- PASS: TestAccBatchJobDefinition_NodePropertiesupdateForcesNewResource (37.68s)
--- PASS: TestAccBatchJobDefinition_tags (45.92s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/batch	47.862s

```

### CODE REVIEW

File: internal/service/batch/job_definition.go [hunk 1]
```diff
@@ -69,9 +70,9 @@ func ResourceJobDefinition() *schema.Resource {
 				ValidateFunc: validName,
 			},
 			"node_properties": {
-				Type:     schema.TypeString,
-				Optional: true,
-				ForceNew: true,
+				Type:          schema.TypeString,
+				Optional:      true,
+				ConflictsWith: []string{"eks_properties"},
```
Comment (drewmullen) [inline internal/service/batch/job_definition.go:75]:
```suggestion
				ConflictsWith: []string{"eks_properties", "container_properties"},
```

File: internal/service/batch/job_definition.go [hunk 2]
```diff
@@ -48,9 +48,10 @@ func ResourceJobDefinition() *schema.Resource {
 				Computed: true,
 			},
 			"container_properties": {
-				Type:     schema.TypeString,
-				Optional: true,
-				ForceNew: true,
+				Type:          schema.TypeString,
+				Optional:      true,
+				ForceNew:      true,
+				ConflictsWith: []string{"eks_properties"},
```
Comment (drewmullen) [inline internal/service/batch/job_definition.go:54]:
```suggestion
				ConflictsWith: []string{"node_properties", "container_properties"},
```

File: internal/service/batch/job_definition_test.go [hunk 3]
```diff
@@ -567,6 +567,77 @@ func TestAccBatchJobDefinition_NodePropertiesupdateForcesNewResource(t *testing.
...
+					resource.TestCheckResourceAttr(resourceName, "name", rName),
+					resource.TestCheckResourceAttr(resourceName, "type", "container"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateVerifyIgnore: []string{
+					"deregister_on_new_revision",
```
Comment (drewmullen) [inline internal/service/batch/job_definition_test.go:597]: can be removed, not included yet 😭

File: internal/service/batch/job_definition.go (hunk 1 above)
Comment (danquack) [reply internal/service/batch/job_definition.go:76]: `TestAccBatchJobDefinition_createTypeContainerWithBothProperties` and `TestAccBatchJobDefinition_createTypeMultiNodeWithBothProperties` fails when I add this.

Comment (drewmullen) [reply internal/service/batch/job_definition.go:76]: Those tests expect failure. Could we just eliminate them?

Comment (danquack) [reply internal/service/batch/job_definition.go:76]: Yeah, let me update the string its looking to match on.

**Review by drewmullen — APPROVED** [review]
LGTM!

**Review by jar-b — APPROVED** [review]
LGTM 🎉 

```console
% make testacc PKG=batch TESTS=TestAccBatchJobDefinition_
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/batch/... -v -count 1 -parallel 20 -run='TestAccBatchJobDefinition_'  -timeout 360m

--- PASS: TestAccBatchJobDefinition_createTypeMultiNodeWithContainerProperties (7.92s)
--- PASS: TestAccBatchJobDefinition_createTypeContainerWithNodeProperties (8.09s)
--- PASS: TestAccBatchJobDefinition_EKSProperties_basic (21.23s)
--- PASS: TestAccBatchJobDefinition_ContainerProperties_EmptyField (21.49s)
--- PASS: TestAccBatchJobDefinition_basic (21.51s)
--- PASS: TestAccBatchJobDefinition_PlatformCapabilities_ec2 (21.68s)
--- PASS: TestAccBatchJobDefinition_PlatformCapabilitiesFargate_containerPropertiesDefaults (24.66s)
--- PASS: TestAccBatchJobDefinition_PlatformCapabilities_fargate (24.76s)
--- PASS: TestAccBatchJobDefinition_propagateTags (26.51s)
--- PASS: TestAccBatchJobDefinition_disappears (27.29s)
--- PASS: TestAccBatchJobDefinition_ContainerProperties_advanced (29.56s)
--- PASS: TestAccBatchJobDefinition_NodeProperties_basic (30.50s)
--- PASS: TestAccBatchJobDefinition_updateForcesNewResource (30.51s)
--- PASS: TestAccBatchJobDefinition_EKSProperties_update (31.89s)
--- PASS: TestAccBatchJobDefinition_NodePropertiesupdateForcesNewResource (32.64s)
--- PASS: TestAccBatchJobDefinition_tags (36.83s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/batch      40.082s
```

**Comment by jar-b** [comment]
Thanks for your contribution, @danquack! Just replaced a couple inline strings with enum value helpers (missing from the AWS SDK).

**Pull request merged**